wurst/util/Flexboxgrid.wurst

```diff
@@ -0,0 +1,366 @@
+	..setWidth(width - 0.001)
+	..setValue(value)
+	..setIcon(icon)
+	..setValueColor(clr)
+	.release()
+
+abstract class Flex // hello, ricardo
+	protected static Flexboxgrid cfbx = null // parent flexboxgrid during draw()
+	protected var width_ = 0.
+	protected var height_ = 1
+	protected var fixed_h = false
+	protected var fixed_w = false
+	protected abstract function draw(int y, real width_, int height_)
+	protected abstract function calc()
+	protected abstract function update()
+
+public interface CellUpdater
+	function update(Cell c)
+
+constant STUB = new Cell()
+public class Cell extends Flex
+	var value = ""
+	var icon = ""
+	var clr = COLOR_WHITE
+	var width = 0.
+	CellUpdater updater = null
+
+	construct()
+	
+	construct(real width_, CellUpdater updater_)
+		width = width_
+		updater = updater_
+		update()
+
+	construct(real width_, string value_, string icon_, colorA clr_)
+		width = width_
+		value = value_
+		icon = icon_
+		clr = clr_
+	
+	ondestroy
+		destroy updater
+
+	function clear()
+		value = ""
+		icon = ""
+		clr = COLOR_WHITE
+		width = 0.
+
+	override function calc()
+		height_ = 1
+		width_ = width
+
+	override function draw(int y, real width, int height)
+		cfbx.nextRowItem(y).preset(width, value, icon, clr)
+		var py = y + 1
+		while py < y + height
+			cfbx.nextRowItem(py).preset(width, "", "", COLOR_WHITE)
+			py++
+
+	override function update()
+		if updater != null
+			updater.update(this)
+
+
+abstract class Container extends Flex
+	Container parent = null
+	abstract function insert(Flex o)
+	function line()
+		Log.error("[Flexboxgrid] called .line() on wrong container type!")
+
+abstract class LinearContainer extends Container
+	let objs = new LinkedList<Flex>
+
+	ondestroy
+		for o in objs
+			destroy o
+		destroy objs
+
+	override function insert(Flex o)
+		objs.add(o)
+
+	override function update()
+		for o in objs
+			o.update()
+
+class Row extends LinearContainer
+	real sum_width_ = 0.
+	override function calc()
+		var max_height_ = 1
+		sum_width_ = 0
+		for o in objs
+			o.calc()
+			max_height_ = max(max_height_, o.height_)
+			sum_width_ += o.width_
+		if not fixed_w
+			width_ = sum_width_
+		if not fixed_h or height_ < max_height_
+			height_ = max_height_
+
+	override function draw(int y, real width, int height)
+		for o in objs
+			o.draw(y, width * o.width_ / sum_width_, height)
+
+class Col extends LinearContainer
+	int sum_height_ = 0
+	override function calc()
+		var max_width_ = 0.
+		sum_height_ = 0
+		for o in objs
+			o.calc()
+			max_width_ = max(max_width_, o.width_)
+			sum_height_ += o.height_
+		if not fixed_w
+			width_ = max_width_
+		if not fixed_h or height_ < sum_height_
+			height_ = sum_height_
+
+	override function draw(int y, real width, int height)
+		var py = y
+		for o in objs
+			o.draw(py, width, o.height_)
+			py += o.height_
+
+		while py < y + height
+			cfbx.nextRowItem(py).preset(width, "", "", COLOR_WHITE)
+			py++
+
+class Table extends Container
```

Comment:
Table class already exists from package Table, rather rename to avoid confusion.